### ISSUE TITLE
A little regression of undesirable line break on upgrading 3.5.8 -> 3.5.9

### ISSUE THREAD

**Issue by danicheg** [scalameta/scalafmt#3309]
In the blaze, we faced an [adding a line break](https://github.com/http4s/blaze/pull/723/files#diff-75708f00aaaed216a2d15f536422faace89e41db2ada2bf708411f8cd9ae77f2R151-R153) when upgrading Scalafmt 3.5.8 -> 3.5.9. 

## Configuration ##
The config remained [the same](https://github.com/http4s/blaze/blob/main/.scalafmt.conf) (besides bumping the version).

## Steps
A simplified example:
```scala
def this(
      responseHeaderTimeout: Duration,
      idleTimeout: Duration,
  ) = this(
    responseHeaderTimeout = responseHeaderTimeout,
    idleTimeout = idleTimeout
  )(F)
```

## Problem
```scala
def this(
      responseHeaderTimeout: Duration,
      idleTimeout: Duration,
  ) = this(
    responseHeaderTimeout = responseHeaderTimeout,
    idleTimeout = idleTimeout
  )(
    F
  )
```

## Expectation
That code should remain unchanged.

Can you please confirm – this behavior is expected or is it a regression?

**Comment by kitbellew** [comment]
> A simplified example:
> 
> ```scala
> def this(
>       responseHeaderTimeout: Duration,
>       idleTimeout: Duration,
>   ) = this(
>     responseHeaderTimeout = responseHeaderTimeout,
>     idleTimeout = idleTimeout
>   )(F)
> ```

does this simplified example actually reproduce the problem for you? `runner.optimizer` sets, by default, `forceConfigStyleOnOffset = 150`, and the non-whitespace distance between `this(` and closing `)` is less than 150 bytes in the simplified version.

for me, this example worked as intended but the longer version (from the linked source) exhibited the problem.

**Comment by danicheg** [comment]
> does this simplified example actually reproduce the problem for you?

Not quite. I added it just to highlight the problem in the linked source.

**Comment by kitbellew** [comment]
> > does this simplified example actually reproduce the problem for you?
> 
> Not quite. I added it just to highlight the problem in the linked source.

that's what i thought. perhaps you could provide an actually reproducible example next time; as i don't know what's causing this behavior, the only way to diagnose is to reproduce and obtain a log of the specific rules which were applied.

**Comment by danicheg** [comment]
> perhaps you could provide an actually reproducible example next time

sure. sorry for that inaccuracy.
